ergotree-ir/src/types/sgroup_elem.rs

```diff
@@ -0,0 +1,73 @@
+use crate::serialization::types::TypeCode;
+
+use super::sfunc::SFunc;
+use super::smethod::MethodId;
+use super::smethod::SMethod;
+use super::smethod::SMethodDesc;
+use super::stype::SType;
+use super::stype_companion::STypeCompanion;
+use super::stype_companion::STypeCompanionHead;
+use super::stype_param::STypeVar;
+use lazy_static::lazy_static;
+
+/// SGroupElement type id
+pub const TYPE_ID: TypeCode = TypeCode::SGROUP_ELEMENT;
+/// GroupElement.getEncoded
+pub const GET_ENCODED_METHOD_ID: MethodId = MethodId(2);
+/// GroupElement.negate
+pub const NEGATE_METHOD_ID: MethodId = MethodId(5);
+
+static S_GROUP_ELEM_TYPE_COMPANION_HEAD: STypeCompanionHead = STypeCompanionHead {
+    type_id: TYPE_ID,
+    type_name: "GroupElement",
+};
+
+lazy_static! {
+    /// GroupElement object type companion
+    pub static ref S_GROUP_ELEM_TYPE_COMPANION: STypeCompanion = STypeCompanion::new(
+        &S_GROUP_ELEM_TYPE_COMPANION_HEAD,
+        vec![
+            &GET_ENCODED_METHOD_DESC,
+            &NEGATE_METHOD_DESC
+        ]
+    );
+}
+
+lazy_static! {
+    static ref GET_ENCODED_METHOD_DESC: SMethodDesc = SMethodDesc {
+        method_id: GET_ENCODED_METHOD_ID,
+        name: "getEncoded",
+        tpe: SFunc::new(
+            vec![SType::SGroupElement],
+            SType::SColl(SType::STypeVar(STypeVar::ov()).into()),
```

Comment:
The result type should be `Coll[Byte]`.

Reply:
`STypeVar` is used when we need to express a polymorphic type (think `Coll.map`, etc.).

Reply:
Fixed.